aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cu

```diff
@@ -106,7 +106,14 @@ SparseTensor _coalesce_sparse_cuda(const SparseTensor& self) {
     values = values.contiguous();
     int64_t stride = c10::multiply_integers(values.sizes().slice(1));
     int warp_size = at::cuda::warp_size();
+#ifdef USE_ROCM
+    const int64_t BATCHING_SEGMENT = 4096;
```

Comment:
@xinyazhang Can we define this constant in the .cuh file so we don't have to repeat the literal value everywhere? Or could we use `gridDim.y` instead of `4096` in aten/src/ATen/native/sparse/cuda/SparseCUDAApplyUtils.cuh?